stdlib/SharedArrays/src/SharedArrays.jl

```diff
@@ -559,6 +559,8 @@ similar(S::SharedArray) = similar(S.s, eltype(S), size(S))
 reduce(f, S::SharedArray) =
     mapreduce(fetch, f, Any[ @spawnat p reduce(f, S.loc_subarr_1d) for p in procs(S) ])
 
+reduce(::typeof(merge), S::SharedArray) =
+    throw(ArgumentError("cannot call `merge` on a `SharedArray`"))
```

Comment:
Why is this needed at all?

Reply:
I guess because of the ambiguity with the previous method?

Reply:
So this PR breaks `reduce(merge, ::SharedArray)`?

Reply:
Ah, right, better change it to:
```julia
reduce(::typeof(merge), S::SharedVector) = invoke(reduce, Tuple{Any,SharedArray}, merge, S)
```
(I know nothing about `SharedArray` so I don't know whether the new optimized method would work or not.)

Reply:
*  `@view`: Ok.

 * `MethodError`: Yes, this is better

 * Ah `invoke` (!) That's the function I mentioned whose name I could not remember. I just threw an error instead, because  "type of SharedArray elements must be bits types".  The chance that a method is written for this case near zero. But, of course it should be allowed.

* Tests: I will add tests. I doubt the code would be covered otherwise.

* Commas: It's my preference to always add a space. I was trying to follow the example of some functions that I saw nearby in the file (IIRC). The insertion of a space or not after a comma in "base/abstractdict.jl" is more or less random. But, sure, I'll do it.